### PR TITLE
fetching user location based on IP

### PR DESCRIPTION
**Pull request by ilonamm**
#28

### CODE REVIEW

**Comment by ilonamm** [comment]
This only adds the location to new users, not sure why 😄 

this is what this looks like in the db.
image

**Comment by ilonamm** [comment]
@batmany13 need to migrate the db on qa, could you give pointers as to the right process? thank you!

this didn't work, committed a migration file but looks like it didn't notice that. does it need to be done locally?

`ilonamooney$ heroku run "mix ecto.migrate" -a askbywho-qa
Running mix ecto.migrate on ⬢ askbywho-qa... up, run.6906 (Free)

11:03:23.521 [info]  Already up`

**Comment by ilonamm** [comment]
Got this to work :)
image
will clean up the code from debugging logs

**Comment by ilonamm** [comment]
This includes also #39

**Comment by ilonamm** [comment]
when you have time please check this @batmany13!

File: lib/askbywhoplugs/remoteipplug.ex [hunk 1]
```diff
@@ -0,0 +1,17 @@
+defmodule Askbywhoplugs.RemoteIPPlug do
```
Comment (batmany13) [inline lib/askbywhoplugs/remoteipplug.ex:1]: this plug name seems redundant.  Also, typically names should be like "remoteip_plug.ex", using "_" to denote a space

File: priv/repo/migrations/20170809102200_emails_location_for_prod.exs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+defmodule Askbywho.Repo.Migrations.EmailsLocationForProd do
+  use Ecto.Migration
+
+  def change do
+    alter table(:emails) do
+      add :latitude, :real
+      add :longitude, :real
+      add :location, :string
```
Comment (batmany13) [inline priv/repo/migrations/20170809102200_emails_location_for_prod.exs:8]: is this the "city"?  Why don't we include the original IP as well?

File: web/views/page_view.ex [hunk 3]
```diff
@@ -59,5 +59,8 @@ defmodule Askbywho.PageView do
     |> raw
   end
 
-
+  def on_qa do
+    IO.puts("environment var is_qa is #{inspect System.get_env("is_qa")}")
```
Comment (batmany13) [inline web/views/page_view.ex]: should use Logger for this type of stuff

File: web/views/page_view.ex [hunk 4]
```diff
@@ -59,5 +59,8 @@ defmodule Askbywho.PageView do
...
+    System.get_env("is_qa") == "yes"
```
Comment (batmany13) [inline web/views/page_view.ex]: Why does there have to be a special "is_qa' env variable?  Couldn't you just check for a specific MIX_ENV?

File: priv/repo/migrations/20170809102200_emails_location_for_prod.exs (hunk 2 above)
Comment (ilonamm) [reply priv/repo/migrations/20170809102200_emails_location_for_prod.exs:8]: was thinking if it is dynamic, does it make sense to save? so the city is nice to see, but saving lat and long as it is more versatile

File: web/views/page_view.ex (hunk 4 above)
Comment (ilonamm) [reply web/views/page_view.ex]: initially was going to do that but now QA thinks it is prod so I would have needed to set up a new environment and seemed there would have been a lot of config to do, this was quicker

File: priv/repo/migrations/20170809102200_emails_location_for_prod.exs (hunk 2 above)
Comment (batmany13) [reply priv/repo/migrations/20170809102200_emails_location_for_prod.exs:8]: ok, but its assuming your geoip is current and getting the right lat/long, just seems like having the original could be useful.  Not a huge deal

File: web/views/page_view.ex (hunk 4 above)
Comment (batmany13) [reply web/views/page_view.ex]: what do you mean "qa thinks it is prod"?  Is QA an actual different server?  If its a diff server, you should use diff settings than prod, so having its own config/qa.env file and then you can follow normal convention for types of environments

**Review by batmany13 — APPROVED** [review]
we need to figure out this "is_qa" thing though

File: priv/repo/migrations/20170809102200_emails_location_for_prod.exs (hunk 2 above)
Comment (ilonamm) [reply priv/repo/migrations/20170809102200_emails_location_for_prod.exs:8]: added as #41

**Pull request merged**